app/helpers/duties_helper.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module DutiesHelper
+  def process_duties(start_date, end_date)
+    result = []
```

Comment:
`start_date.upto(end_date).map { |day| Place.all.map { |place| process_day_place(day, place) } }`

Will something like this work?